webv2/updateTableSchema/renameColumn.go

```diff
@@ -0,0 +1,316 @@
+	for i, _ := range sp.Fks {
+
+		renameColumnNameInForeignkeyTableSchema(Conv, sp, i, colName, newName)
+	}
+
+	for _, sp := range Conv.SpSchema {
+
+		for j := 0; j < len(sp.Fks); j++ {
+			if sp.Fks[j].ReferTable == table {
+				renameColumnNameInForeignkeyReferTableSchema(Conv, sp, sp.Name, colName, newName)
+			}
+
+		}
+
+	}
+
+	// update interleave table relation
+	isParent, parentschemaTable := IsParent(table)
+
+	if isParent {
+		renameColumnNameInParentTableSchema(Conv, parentschemaTable, colName, newName)
+
+	}
+
+	childschemaTable := Conv.SpSchema[table].Parent
+
+	if childschemaTable != "" {
+
+		renameColumnNameInchildTableSchema(Conv, childschemaTable, colName, newName)
+
+	}
+
+}
+
+// renameColumnNameInCurrentTableSchema renames given column in Current Table Schema.
+func renameColumnNameInCurrentTableSchema(Conv *internal.Conv, sp ddl.CreateTable, table string, colName string, newName string) {
+	// step I
+	sp = renameColumnNameInSpannerColDefs(sp, colName, newName)
+
+	// step II
+	sp = renameColumnNameInSpannerPK(sp, colName, newName)
+
+	// step III
+	sp = renameColumnNameInSpannerSecondaryIndex(sp, colName, newName)
+
+	// step IV
+	sp = renameColumnNameInSpannerForeignkeyColumns(sp, colName, newName)
+
+	// step V
+	sp = renameColumnNameInSpannerForeignkeyReferColumns(sp, colName, newName)
+
+	// step VI
+	sp = renameColumnNameInSpannerColNames(sp, colName, newName)
+
+	// step VII
+	renameColumnNameInSpannerSchemaIssue(table, colName, newName, Conv)
+
+	// step VIII
+	renameColumnNameInToSpannerToSource(table, colName, newName, Conv)
+
+	Conv.SpSchema[table] = sp
+
+}
+
+// renameColumnNameInForeignkeyTableSchema renames given column in Foreignkey Table Schema.
+func renameColumnNameInForeignkeyTableSchema(Conv *internal.Conv, sp ddl.CreateTable, index int, colName string, newName string) {
+
+	relationTable := sp.Fks[index].ReferTable
+
+	relationTableSp := Conv.SpSchema[relationTable]
+
+	relationTableSp = renameColumnNameInSpannerColNames(relationTableSp, colName, newName)
+	relationTableSp = renameColumnNameInSpannerColDefs(relationTableSp, colName, newName)
+	relationTableSp = renameColumnNameInSpannerPK(relationTableSp, colName, newName)
+	relationTableSp = renameColumnNameInSpannerSecondaryIndex(relationTableSp, colName, newName)
+	relationTableSp = renameColumnNameInSpannerForeignkeyColumns(relationTableSp, colName, newName)
+	relationTableSp = renameColumnNameInSpannerForeignkeyReferColumns(relationTableSp, colName, newName)
+
+	renameColumnNameInToSpannerToSource(relationTable, colName, newName, Conv)
+	renameColumnNameInSpannerSchemaIssue(relationTable, colName, newName, Conv)
+
+	Conv.SpSchema[relationTable] = relationTableSp
+
+}
+
+// renameColumnNameInForeignkeyReferTableSchema renames given column in Foreignkey Refer Table Schema.
+func renameColumnNameInForeignkeyReferTableSchema(Conv *internal.Conv, sp ddl.CreateTable, table string, colName string, newName string) {
```

Comment:
No need of this function , reuse - renameColumnNameInCurrentTableSchema(). Rename the method to solve both the purposes.

Reply:
Updated